pytorch_lightning/trainer/trainer.py

```diff
@@ -876,9 +866,25 @@ def _run(self, model: 'pl.LightningModule') -> Optional[Union[_EVALUATE_OUTPUT,
         # plugin will setup fitting (e.g. ddp will launch child processes)
         self._pre_dispatch()
 
+        # restore modules after setup
+        self.checkpoint_connector.resume_start()
```

Comment:
@awaelchli I believe you had a reason for having them before setup right ?

Reply:
Yep you're right @tchaton this logic will need to be moved back, as this is necessary for the user to load weights before modules are wrapped via Fully Sharded `configure_sharded_model`.

Going to be a tough one to resolve without moving this as deepspeed is required to be initialized to load the model, but going to see what I can do